core/src/main/java/org/apache/calcite/sql/dialect/OracleSqlDialect.java

```diff
@@ -188,4 +191,13 @@ public OracleSqlDialect(Context context) {
       }
     }
   }
+
+  @Override public void unparseOffsetFetch(SqlWriter writer, @Nullable SqlNode offset,
+       @Nullable SqlNode fetch) {
+    // majorVersion in SqlDialect.EMPTY_CONTEXT is -1 by default
+    if (this.majorVersion != -1 && this.majorVersion < 12) {
+      throw new RuntimeException("Lower Oracle version(<12) doesn't support offset/fetch syntax!");
+    }
+    unparseFetchUsingAnsi(writer, offset, fetch);
```

Comment:
`super.unparseOffsetFetch(writer, offset, fetch))` is better for me.

Reply:
Make sense, I've updated the PR.